From: SeongJae Park <sj@kernel.org>
Date: Sat, 12 Apr 2025 13:16:48 -0700
Subject: [PATCH] ==== addr_unit (arm32 lpae) ====

mm/damon: support address space larger than damon-core address space

DAMON core layer has its own address space that uses 'unsigned long' for
handling addresses.  The underlying DAMON operations set implementation
translates the core layer address to the real address space that it is
working for.   For example, vaddr and fvaddr translates the core layer
addresses to virtual addresses.

However, all operations set layer simply assumes the core layer address
will be same to those for the given address space.  Since the core layer
address uses 'unsigned long', it is ok for virtual addresses.  In case
of physical address space, however, there can be some cases where
'unsigned long' cannot handle whole address.  For example, 32bit arm
machines with large physical address extension (LPAE) is such a case[1].
Arm with LPAE is the only reported use case, but similar cases could
happen in future.

Add a new core layer parameter called 'addr_unit'.  Operations set layer
can translate a core layer address to the real address by multiplying
the parameter value to the core layer address.  Support of the parameter
is up to each operations layer implementation, though.  For example,
operations set implementations for virtual address space can simply
ignore the parameter.  Add the support on paddr, which is the DAMON
operations set implementation for the physical address space, as we have
a clear use case for that.

Patches Sequence
================

The first patch implements the new parameter on DAMON kernel API.

Following five patches (patch 2-6) incrementally add support of the new
parameter on paddr operations set, for monitoring operations and DAMOS
operations for each DAMOS actions.

The seventh patch implements DAMON sysfs ABI for letting users control
the parameter.

Final three patches (patches 8-10) documents the new ABI on design,
usage, and ABI documents, respectively.

[1] https://lore.kernel.org/20250408075553.959388-1-zuoze1@huawei.com

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_kfx5iq0v | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_kfx5iq0v

diff --git a/hkml_cv_bogus/hkml_cv_bogus_kfx5iq0v b/hkml_cv_bogus/hkml_cv_bogus_kfx5iq0v
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

